dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -202,12 +203,34 @@ func startSyncWriter(q queue.Queue) queue.Queue {
 }
 
 func (cw *ClientWriter) sendSuccessPiece(piece *Piece, cost time.Duration) {
-	cw.api.ReportPiece(piece.SuperNode, &types.ReportPieceRequest{
+	reportPieceRequest := &types.ReportPieceRequest{
 		TaskID:     piece.TaskID,
 		Cid:        cw.cfg.RV.Cid,
 		DstCid:     piece.DstCid,
 		PieceRange: piece.Range,
-	})
+	}
+
+	var retry = 0
+	var maxRetryTime = 3
+	for {
+		_, err := cw.api.ReportPiece(piece.SuperNode, reportPieceRequest)
+		if err == nil {
+			if retry > 0 {
+				logrus.Warnf("success to report piece with request(%+v) after retrying (%d) times", reportPieceRequest, retry)
+			}
+			break
+		}
+		if retry >= maxRetryTime {
+			logrus.Errorf("failed to report piece to supernode with request(%+v) even after retrying max retry time", reportPieceRequest)
+			break
+		}
+
+		sleepTime := time.Duration(rand.Intn(1000)+500) * time.Millisecond
```

Comment:
The time to sleep is a bit long. It's better to decrease the interval time of first retry. How about [50, 500]ms?

Reply:
DONE.